core/src/win32/plugins/filed/mssqlvdi-fd.cc

```diff
@@ -958,6 +958,14 @@ static void SetAdoConnectString(PluginContext* ctx)
   PoolMem ado_connect_string(PM_NAME);
   plugin_ctx* p_ctx = (plugin_ctx*)ctx->plugin_private_context;
 
+  // If no explicit instance name given usedthe DEFAULT_INSTANCE name.
+  if (!p_ctx->instance) { p_ctx->instance = strdup(DEFAULT_INSTANCE); }
+
+  // If no explicit server address given use the DEFAULT_SERVER_ADDRESS.
+  if (!p_ctx->server_address) {
+    p_ctx->server_address = strdup(DEFAULT_SERVER_ADDRESS);
+  }
```

Comment:
I'm not sure about this, but i think this is done somewhere else.  I think its kinda unexpected that a function called `SetAdoConnectString` actually also sets the server_address and the instance.

I had the two following ideas:
1. since `parse_plugin_definition` kinda acts like a constructor we could add 
``` 
  SetStringIfNull(&p_ctx->server_address, DEFAULT_SERVER_ADDRESS);
  SetStringIfNull(&p_ctx->instance, DEFAULT_INSTANCE);
``` 
  just before returning.  This has the obvious downside that it can change the behaviour of that function if it ever gets called twice on the same context.   I hope this does not actually happen!

2. Add `get_server_name` and `get_instance` functions that return the default if the real ones are set to null.  This has the advantage that we do not need to needlessly copy the default values.  In that case the functions should return a `const char *` just to be safe.  This comes with the downside of breaking if the rest of the code expects `p_ctx->instance` to not be zero.

What do you think about these suggestions ?

Reply:
Looking at the code (in this file at least) it seems that it would be pretty easy to replace accesses with calls to the accessor function instead. So maybe it could work.